miner/worker.go

```diff
@@ -1114,11 +1123,11 @@ func (w *worker) commitWork(interrupt *int32, noempty bool, timestamp int64) {
 	// Set the coinbase if the worker is running or it's required
 	var coinbase common.Address
 	if w.isRunning() {
-		if w.coinbase == (common.Address{}) {
+		coinbase = w.etherbase()
+		if coinbase == (common.Address{}) {
```

Comment:
But - this still requires an etherbase, right? If the work-package from CL contains an etherbase, this shouldn't be needed, should it?

Reply:
`commitWork` is only for legacy block generation.
`generateWork` is for post-merge mode.